pyspecdata/DCCT_function.py

```diff
@@ -235,32 +268,34 @@ def draw_span(
         allow_for_text=allow_for_text_default,
         allow_for_ticks=allow_for_ticks_default,
     ):
-        x1, y1 = ax1.transAxes.transform(r_[0, 0.95])
-        x2, y2 = ax2.transAxes.transform(r_[0, 0.05])
-        x1 -= allow_for_ticks
-        x_text = x1 - allow_for_text
-        x2 -= allow_for_ticks
-        # following line to create an offset for different dimension labels
         label_spacing = this_label_num * label_spacing_multiplier
-        x1, y1 = fig.transFigure.inverted().transform(r_[x1 - label_spacing, y1])
-        x_text, _ = fig.transFigure.inverted().transform(r_[x_text - label_spacing, 0])
-        x2, y2 = fig.transFigure.inverted().transform(r_[x2 - label_spacing, y2])
+        # {{ transform from axes coords to figure coords for respective axes
+        ax1_to_figure = ax1.transAxes + fig.transFigure.inverted()
```

Comment:
I don't understand why you need this. I see where it is implemented, but I do not understand why any spacing, etc. that you need to specify between the plots and what not can't be specified using figure coordinates.

Phrased differently, I don't see any need for any coordinate systems aside from the following:

- The blended transform where the Y values are just figure coordinates, and the X values are in display units, with the to the bottom left corner of the bottom left figure. (Achieved with a scaled translation.)
    - this coordinate system is used to specify all of the decoration on the left-hand side aside from the arrows on the bottom
- a coordinate system where all values are in display units with the origin set to the bottom left corner of the bottom left figure. This is achieved using scaledtranslation, and is the coordinate system that is used for the X coordinates of the previous coordinate system. So, this coordinate system should actually be defined first, and then input into the blended transform factory.
   - as clarified in the Slack conversation, this is used for the arrows

Reply:
This applies to the definition of all these extra coordinate systems like `ax1_to_figure`, etc

Reply:
[see here](https://github.com/jmfrancklab/pyspecdata/pull/65#discussion_r1828541716)